draft-jholland-quic-multicast.md

```diff
@@ -287,55 +303,67 @@ The same stream ID may be used in both one or more multicast channels and the un
 
 The values used for unicast flow control cannot be used to limit the transmission rate of a multicast channel because a single client with a low MAX_STREAM_DATA or MAX_DATA value that did not acknowledge receipt could block many other receivers if the servers had to ensure that channels responded to each client's limits.
 
-Instead, clients advertise resource limits that the server is responsible for staying within via MC_LIMITS ({{client-limits-frame}}) frames and their initial values from the transport parameter ({{transport-parameter}}).
-The server advertises the expected maxima of the values that can contribute toward client resource limits within a channel in an MC_ANNOUNCE ({{channel-announce-frame}}) frame.
+Instead, clients advertise resource limits via MC_LIMITS ({{client-limits-frame}}) frames and their initial values from the transport parameter ({{transport-parameter}}).
+The server is responsible for keeping the client within its advertised limits, by ensuring via MC_JOIN and MC_LEAVE frames that the set of channels the client is asked to be joined to will not, in aggregate, exeed the client's advertised limits.
+The server also advertises the expected maxima of the values that can contribute toward client resource limits within a channel in an MC_ANNOUNCE ({{channel-announce-frame}}) frame, and the client also ensures that the set of channels it's joined to does not exceed its limits, according to the advertised values.
+The client also monitors the packets received to ensure that channels don't exceed their advertised values, and leaves channels that do.
```

Comment:
The description of Max Rate in the MC_ANNOUNCE frame specifies a 5s window for max rate, but I think we should probably explain that in the flow control section too, yes.  I think it does probably have to be part of the spec (at least as RECOMMENDED or SHOULD or something), otherwise different implementations will not necessarily interoperate well.